specification/security/resource-manager/Microsoft.Security/preview/2024-05-01-preview/alerts.json

```diff
@@ -0,0 +1,1246 @@
+        "description": "Update the alert's state",
+        "operationId": "Alerts_UpdateResourceGroupLevelStateToInProgress",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/AscLocation"
+          },
+          {
+            "$ref": "#/parameters/AlertName"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Security/locations/{ascLocation}/alerts/default/simulate": {
+      "post": {
+        "x-ms-examples": {
+          "Simulate security alerts on a subscription": {
+            "$ref": "./examples/Alerts/SimulateAlerts_example.json"
+          }
+        },
+        "tags": [
+          "Alerts"
+        ],
+        "description": "Simulate security alerts",
+        "operationId": "Alerts_Simulate",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/AscLocation"
+          },
+          {
+            "$ref": "#/parameters/AlertSimulatorRequestBody"
+          }
+        ],
+        "x-ms-long-running-operation": false,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "original-uri"
+        },
+        "responses": {
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Alert": {
+      "type": "object",
+      "description": "Security alert",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "describes security alert properties.",
+          "$ref": "#/definitions/AlertProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "AlertEntity": {
+      "type": "object",
+      "additionalProperties": {},
```

Comment:
>additionalProperties": {} [](http://example.com/codeflow?start=7&length=25)

"additionalProperties" is an ARM anti-pattern and generally not allowed. Since your previous API version had it already, you aren't actually adding it in the API version, so it's grandfathered in. However, when additionalProperties is provided, it requires a type definition, so removing the type is not allowed.

Reply:
so if I understand you correctly, the fix is to change it to:
"additionalProperties": {
        "type": "object"
      },
      
correct?

Reply:
I'll play it back to be sure I understand the rationale here. This is a change from the current swagger to remove the type from the additionalProperties definition (IIUC). Are you saying that your current swagger is incorrectly modeling this API, and this update is to fix it?

If that's the case, what's different between your your current swagger and the API behavior? Note that "entities" is an array of objects. Is the issue that the objects in that array are completely unconstrained JSON (IOW, their properties could be any name and any type)?